### DISCUSSION TITLE
Clover Bootloader freezes on my PC

### DISCUSSION THREAD

**Discussion by usbeehu** [CloverHackyColor/CloverBootloader#652] [General]
I'm using r5156 from an USB drive using the iso file and Ventoy.
I have a Gigabyte GA-880GMA-USB3 motherboard with EFI support, a Phenom II CPU and 32 GB of DDR3 RAM.
When I select Clover from Ventoy's menu, it loads up, showing the main menu, but freezes immediately.
I tried to put the iso file to a pendrive with Balena Etcher but it won't load at all.

**Comment by Morbius01** [comment]
I am not sure whether Ventoy could be the cause of your problem.
What I can tell you is that I have a main board even older that yours (non UEFI), and there is no way I can use v.5156 (or any of the most recent versions of Clover) : my PC would freeze even before I reach Clover's main menu !
And several people using legacy systems have reported too being unable to use the recent versions of Clover.

I can only suggest you to read what I did to successfully create a Clover key, and then install several OSs on my non-bootable NVMe SSD.
For this, I used an earlier version of Clover, v.4961 and the BD Utility tool.
You may like to try it. Who knows, it might work for you too.
See [here](https://github.com/CloverHackyColor/CloverBootloader/discussions/491).

**Reply by usbeehu** [reply]
Hi! Thank you for your suggestion. I donwloaded your version, but sadly it freezes too. What should I do?

**Reply by Morbius01** [reply]
1. Which application did you use to create the Clover key (which I assume is the v.4961 since you said you used "my version") ?

2. Did you use Ventoy when trying to load this version of Clover ?

3. When exactly does it freeze ?  What is displayed on your screen just before it freezes ?  Do you get to the main Clover menu, and then it freezes, or does it freeze before ?

4. Which storage device do you intend to use Clover with ?  For instance, is it an M.2 NVMe SSD installed on a PCIe adapter ?  What brand and model ?  Please be as precise as possible.

5. Which OS(s) do you wish to install on this device with Clover ?

**Reply by usbeehu** [reply]
I used 4961, yes. I created the boot device with the included tool. I was make sure to use the local file instead of online one. It displays the main screen, then sometimes I'm able to move within the menu with the arrow keys, sometimes not, but anyway it freezes. I want to use an NVMe SSD which is the original drive from my Macbook Air (so I wouldn't have to buy another SSD). Which is a Samsung one. I can't name the exact model and I can't find it in my browser history.
I wish to install Windows 10 but as of now it is irrelevant because I can't reach that point where I could load an actual installer.

**Reply by Morbius01** [reply]
Do you have a PCIe adapter for the NVMe SSD ?
You did not explicitly say how you want to install your SSD on your Gigabyte main board...


So, let me first try to recapitulate what you did.

You created a Clover key using BD Utility (presumably, the one I included in the package I posted in the other discussion).
Then you booted from this Clover key, and you get to the main Clover menu screen.
Am I right until now ?

If I am right, congratulations !
You have successfully created a Clover key which boots on your system !
(you may like to know that some others have never reached that step...)

Then, you say that you can move on this menu with the arrows.
(the mouse pointer is inactive for some reason, this is normal)
But it freezes at some point, although you did not indicate what you did before your PC freezes...

Now, let me suggest what you can do next.
First, I assume that you have a Windows 10 Installer on another USB drive, which you may have a created with the Rufus tool on another Windows environment. (Rufus is highly recommended when preparing a  Windows 10 Installer for an installation with Clover)

--->

First, you need to do the following on your newly created Clover USB key, if you have not done it yet :
Copy the ‘EFI/CLOVER/drivers-Off/drivers64UFI/NvmExpressDxe-64.efi’ driver to:
• 'EFI/CLOVER/drivers64’
and to:
• ‘EFI/CLOVER/drivers64UEFI’ 

Then :
Shut down your PC.
Insert your newly created Clover key into a USB port, but **not** the Windows 10 Installer key.
Start your PC.
When you get to the Clover main menu, insert your Windows 10 Installer USB drive into another USB port. Do **not** remove the Clover key !
Then, navigate with the right arrow key to the last entry ‘Exit Clover’, and then, navigate with the down arrow key and select :
Exit Clover → Boot Maintenance Manager → Boot from file → Windows 10 USB drive → \<efi> → \<boot> → bootx64.efi
Can you do this ?
**Describe what happens then.**
Normally, this should start the Windows installation process.

**Reply by usbeehu** [reply]
Sorry for the missing infos. Please ask so I could know what infos are relevant. So yes, I have a PCIe - M.2 adapter an an M.2 - Apple proprietary adapter too. With this setup both Linux and Windows can see that drive, but I cannot set it as a boot drive since the motherboard cannot detect it.

> Am I right until now ?

Exactly. I made the clover key using that tool you put into the zip file, then booted from that pendrive.

Okay, I see: There are two scenarios that could happen based on my attempts. First, after I see the main menu, I do nothing. Then after a while I press arrow keys, but nothing happens. But If I press the arrow keys right after the menu shows up, sometimes I able to navigate in the menu, but then it freezes.

Now: I put the .efi file to these two folders. The problem remains the same. It freezes after the display of the main menu.

**Reply by Morbius01** [reply]
It is really strange that Clover freezes when you try to navigate on the main menu with the arrows.
Can you navigate to the last entry ‘Exit Clover’ and then select it ?

If you can't, perhaps you should try to create your Clover key again with BD Utility :
Start BDUtility.exe.
Click Options → Configuration.
Under Clover Bootloader Source:
. Local and CloverISO-4961.tar.lzma should be selected and Rev: should indicate 4961.
Under Format Options / Boot records:
. Clover should be selected. The Format option should be FAT32 or FAT32-LBA.
Under Multi Partitioning:
. Boot Partition should be selected with a size of 200 MB.
Click OK.
Insert a USB drive. It should appear in Destination Disk.
Click Format, and then OK. The Clover key creation process starts.
Wait until the ‘All Done’ message is displayed (it can take a few minutes…).
The Clover key has been created, and you can close BDUtility.
Of course, you need to copy again the NvmExpressDxe-64.efi driver into the 2 folders.

Then, try again to start your PC with the Clover key and move to last entry on the main menu.
Can you do it ?


Another question.
When you do all this, are you using Ventoy ?

**Reply by usbeehu** [reply]
Still no luck. I did it step by step. I also tried a different pendrive. It's the same result. When I'm quick enough I can navigate in the menu as previously, and if I select continue it simply reloads the main menu, then freezes. I think I should give up at this point.
I don't use Ventoy, it only works with ISOs.

**Reply by Morbius01** [reply]
Frankly, I have never heard before of anyone having your problem : freeze in the main Clover menu !
Usually, it freezes when you select one of the 'wrong' icons. Not when you are in the main menu and do nothing !


1. "and if I select continue it simply reloads the main menu"
what do you mean by 'select continue' ?
There is no 'continue' entry at all in the main Clover menu for version 4961 !
So I am just wondering what menu you have... In fact, I would like to see a copy of your screen before it freezes...

2. Does it freeze when you do nothing, meaning when you don't navigate on the main Clover menu ?
Or does it freeze after you select an icon and press ENTER ?

3. Can you reach the last entry  ‘EXIT’ and select it (press ENTER), or does it freeze before ?
This is important !  Because this is the way you install an OS using this version of Clover.

4. Are you in the UEFI mode, or in the Legacy (non UEFI) mode ? (since you mentioned that your BIOS offers both options) ?


If you want to give up, it is your decision. But you have a system similar to mine : a Gigabyte main board with a Phenom II CPU.
Mine is even older since it is UEFI not compatible.
There are a few more things I can think of that we have not tried yet...
It's up to you.

**Reply by usbeehu** [reply]
It freezes either way, when I do nothing or when I do anything. It doesn't matter. The first option in the menu is "Continue". I will able to send a screenshot later. I'm using UEFI mode. For some reason, my motherboard allows non-UEFI mode for optical drive only.

**Reply by Morbius01** [reply]
Yes. Please send a screenshot of your Clover menu.
I am curious of what it looks like.

And you did not answer Question 3. Can you reach the last entry 'EXIT' ?

**Reply by usbeehu** [reply]
![IMG_4260](https://github.com/CloverHackyColor/CloverBootloader/assets/23268794/21301242-0751-4ca1-8f06-e5fc8ee083e9)

This is the only menu I can access. I was assuming this is the main menu because of this.

**Reply by Morbius01** [reply]
This is not the main Clover menu.
This is the secondary menu that I get after selecting 'EXIT' from the main menu.
Anyway, on this menu, can you select 'Boot Maintenance Manager' ?

And also, can you list all the storage devices you have in your PC ?
Earlier, you mentioned a Samsung NVMe SSD on a PCIe adapter. Is there anything else ? Like any HDD ?

Regarding your NVMe SSD, what does it contain at the moment ?
And what kind of partition table does it have ?  MBR or GPT ?

**Reply by usbeehu** [reply]
I can’t select basically any menu, because it freezes, and I’m not that quick.

I have two SATA HDD through an SATA PCI card but it is unusable with EFI mode (another issue I have to address but it’s a different topic).

The SSD is formatted (exfat or ntfs - I don’t remember) and empty.

**Reply by Morbius01** [reply]
It looks like there is something in your PC configuration which Clover does not like, and makes it freeze.
But right now, I am puzzled about why you don't get to the main Clover menu.

I suggest you do the following :

- Erase everything on your SSD. Then create a **GPT** partition table. But do not format it.
This can be done easily with the GParted utility on any Linux version. (you said earlier that you can see your SSD under Linux, so I assume that you have a live version of Linux on a USB drive, such as Ubuntu or Linux Mint)
If you need further help on how to do this, just ask.
[ I think that you can also create a partition table on Windows using Diskpart or Disk Management.
I find it easier to do it on Linux with GParted ]

- Disconnect your two HDDs

- Restart your PC with the Clover key inserted. (and, of course, the NVMe SSD plugged on its PCIe slot !)

Any results ?

**Reply by usbeehu** [reply]
I did every step exactly as you wrote, but no luck. I even removed the SATA card to make sure. The SSD is perfectly visible in line Linux and I can manage it with GParted with no issue so it can't really has any defect.
There is one thing I may try. There is a newer beta firmware for this motherboard, I wonder if I could try that safely.

**Reply by Morbius01** [reply]
Could you try Clover version 5070 ?

Extract the CloverISO-5070.tar.lzma file from the attached ZIP folder.
Move it to the folder where BDUtility.exe is.
Start BDUtility.
Options > Configuration.
In Local, select  CloverISO-5070.tar.lzma. Click OK.
Insert a USB drive. Select it.
Click Format.

After the Clover key is created, copy the **'\EFI\CLOVER\drivers\off\NvmExpressDxe.efi'** driver to :
- **\EFI\CLOVER\drivers\BIOS**
       and
- **\EFI\CLOVER\drivers\UEFI**

Restart and boot from this new Clover key.

Results ?
Do you get the main Clover Configuration Manager menu ?

[CloverISO-5070.tar.zip](https://github.com/CloverHackyColor/CloverBootloader/files/13732818/CloverISO-5070.tar.zip)
